pyfixest/estimation/decomposition.py

```diff
@@ -0,0 +1,390 @@
+            if not isinstance(values, list):
+                raise TypeError("Values in combine_covariates_dict must be lists.")
+            for v in values:
+                if v not in self.mediator_names:
+                    raise ValueError(f"{v} is not in the mediator names.")
+
+        # Check for overlap in values between different keys
+        all_values = {
+            k: set([v] if isinstance(v, str) else v)
+            for k, v in self.combine_covariates_dict.items()
+        }
+        for key1, values1 in all_values.items():
+            for key2, values2 in all_values.items():
+                if key1 != key2 and values1 & values2:
+                    overlap = values1 & values2
+                    raise ValueError(f"{overlap} is in both {key1} and {key2}.")
+
+    def fit(self, X: np.ndarray, Y: np.ndarray, store: bool = True):
+        "Fit Linear Mediation Model."
+        if store:
+            self.X = X
+            self.N = X.shape[0]
+
+            self.X1 = self.X[:, ~self.mask]
+            self.X1 = np.concatenate([np.ones((self.N, 1)), self.X1], axis=1)
+            self.names_X1 = ["Intercept", self.param]
+            self.param_in_X1_idx = self.names_X1.index(self.param)
+
+            self.X2 = self.X[:, self.mask]
+            self.Y = Y
+
+            results = self.compute_gelbach(
+                X1=self.X1,
+                X2=self.X2,
+                Y=self.Y,
+                X=self.X,
+                agg_first=self.agg_first,
+            )
+
+            (
+                self.direct_effect,
+                self.beta_full,
+                self.beta2,
+                self.beta2_sparse,
+                self.contribution_dict,
+            ) = results
+
+            # Prepare cluster bootstrap if relevant
+            self.X_dict = {}
+            self.Y_dict = {}
+
+            if self.unique_clusters is not None:
+                for g in self.unique_clusters:
+                    cluster_idx = np.where(self.cluster_df == g)[0]
+                    self.X_dict[g] = self.X[cluster_idx]
+                    self.Y_dict[g] = self.Y[cluster_idx]
+
+            return self.contribution_dict
+
+        else:
+            # need to compute X1, X2 in bootstrap sample
+            X1 = np.concatenate([np.ones((X.shape[0], 1)), X[:, ~self.mask]], axis=1)
+            X2 = X[:, self.mask]
+
+            results = self.compute_gelbach(
+                X1=X1,
+                X2=X2,
+                Y=Y,
+                X=X,
+                agg_first=self.agg_first,
+            )
+
+            _, _, _, _, contribution_dict = results
+
+            return contribution_dict
+
+    def bootstrap(self, rng: np.random.Generator, B: int = 1_000, alpha: float = 0.05):
+        "Bootstrap Confidence Intervals for Total, Mediated and Direct Effects."
+        self.alpha = alpha
+        self.B = B
+
+        if self.nthreads != 1:
+            try:
+                from joblib import Parallel, delayed
```

Comment:
add this as a dependency? joblib is pretty close to base (you can also use [multiprocessing](https://docs.python.org/3/library/multiprocessing.html) from the standard library although it is a bit more clunky).

Reply:
Yep, will do, a dependency with "no mandatory dependencies besides Python" sounds pretty reasonable to add =)